files/default/ext_auth_files/pcluster_dcv_ext_auth.py

```diff
@@ -0,0 +1,362 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import print_function
+from future.backports import OrderedDict
+from future.backports.http.server import BaseHTTPRequestHandler, HTTPServer
+from future.backports.socketserver import ThreadingMixIn
+from future.backports.urllib.parse import parse_qsl, urlparse
+
+import hashlib
+import json
+import os
+import random
+import re
+import shutil
+import ssl
+import string
+import subprocess
+import sys
+from datetime import datetime, timedelta
+from pwd import getpwuid
+
+import argparse
+
+from retry.api import retry_call
+
+USER_CREATED_FILE_DIR = "/var/run/exthauth"
```

Comment:
I'd rename to /run/parallelcluster/dcv_ext_auth

We have to create our subfolder for parallelcluster since in future we could have more tool using it.
We should always have a prefix to have a clear idea of the applications is creating a file in /var/run


Then we should use /run and not /var/run: https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch05s13.html